homeassistant/components/v2c/diagnostics.py

```diff
@@ -0,0 +1,40 @@
+"""Diagnostics support for V2C."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+from homeassistant.components.diagnostics import async_redact_data
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST
+from homeassistant.core import HomeAssistant
+from homeassistant.util.json import json_loads
+
+from .const import DOMAIN
+from .coordinator import V2CUpdateCoordinator
+
+CONF_TITLE = "title"
+
+TO_REDACT = {
+    CONF_HOST,
+}
+
+
+async def async_get_config_entry_diagnostics(
+    hass: HomeAssistant, entry: ConfigEntry
+) -> dict[str, Any]:
+    """Return diagnostics for a config entry."""
+    coordinator: V2CUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    if TYPE_CHECKING:
+        assert coordinator.evse
+
+    coordinator_data = await coordinator.evse.get_data()
```

Comment:
The coordinator data is already present in `coordinator.data` right?

Reply:
It is, but after the failed mock (previous review commend) I realised this is a better solution - it will force a refresh which is actually more useful during debug

Reply:
Sure but then you are not creating a diagnostic from the current state. If something goes wrong at a given moment and you want to capture that, snapshotting would refresh the lib and you would get new data that maybe doesn't have the same issue.

Reply:
The important aspect is to capture the differences between the `raw_data` and the `TrydanData` 

The coordinator also updates quite often, it will be very difficult to time a diagnostic download with an ephemeral error.

But if necessary I can change back

Reply:
But with what is raw_data filled then?

Reply:
raw_data is literally the JSON coming out of the EVSE (the physical device)

The objective is to detect whenever the manufacturer changes the JSON and be able to easily provide an update to the integration

Reply:
I agree with @joostlek : we prefer current coordinator data than a fresh one.

Plus, if API changes you will still see it 😌

Reply:
reverted